dds/DCPS/AddressCache.h

```diff
@@ -222,16 +216,41 @@ class AddressCache {
   {
     ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
     const typename IdMapType::iterator pos = id_map_.find(val);
+    KeySet keys;
     if (pos != id_map_.end()) {
-      for (typename KeyVec::iterator it = pos->second.begin(), limit = pos->second.end(); it != limit; ++it) {
-        map_.erase(*it);
-      }
+      keys.swap(pos->second);
       id_map_.erase(pos);
     }
+
+    for (typename KeySet::iterator key = keys.begin(), limit = keys.end(); key != limit; ++key) {
+      map_.erase(*key);
+
+      // Undo insert_ids(key)
```

Comment:
Ah, yeah, I see the problem now. Removing one GUID's `id_map_` entry still potentially leaves ejected keys in other GUID's `id_map_` entry. It doesn't impact cache correctness, but those keys could potentially stick around for a long time. The need to remove the keys here makes me think it really does need to be a set (and not just a sorted vector or whatever).

Reply:
Yea, if there's a ever-changing set of remote participants, the vectors under the local GUIDs grow without limit because I don't think there's anyway for them to be removed.